cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -961,3 +972,95 @@ func (ht *hcsTask) ProcessorInfo(ctx context.Context) (*processorInfo, error) {
 		count: ht.host.ProcessorCount(),
 	}, nil
 }
+
+// Add mount as vSMB share to the UVM at the given destination path
+func (ht *hcsTask) addMountToUVM(ctx context.Context, src string, dst string, isRO bool) (string, error) {
+	options := ht.host.DefaultVSMBOptions(isRO)
+	vsmbShare, err := ht.host.AddVSMB(ctx, src, options)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to add mount as vSMB share to UVM")
+	}
+	defer func() {
+		if err != nil {
+			_ = vsmbShare.Release(ctx)
+		}
+	}()
+
+	sharePath, err := ht.host.GetVSMBUvmPath(ctx, src, isRO)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get vsmb path")
+	}
+	return sharePath, nil
+}
+
+func (ht *hcsTask) requestAddContainerMount(ctx context.Context, resourcePath string, settings interface{}) error {
+	modification := &hcsschema.ModifySettingRequest{
+		ResourcePath: resourcePath,
+		RequestType:  guestrequest.RequestTypeAdd,
+		Settings:     settings,
+	}
+	return ht.c.Modify(ctx, modification)
+}
+
+func isMountTypeSupported(mountType string) bool {
+	// currently we only support mounting of host volumes/directories
+	return mountType == ""
+}
+
+func (ht *hcsTask) updateWCOWContainerMount(ctx context.Context, data interface{}, annotations map[string]string) error {
+	// Hcsschema v2 should be supported
+	if osversion.Build() >= osversion.RS5 {
+		if sv := schemaversion.DetermineSchemaVersion(nil); !schemaversion.IsV21(sv) {
+			return fmt.Errorf("hcsschema v2 unsupported")
+		}
+	}
+	resources, ok := data.(*ctrdtaskapi.ContainerMount)
```

Comment:
don't we already know that the data type is `ContainerMount` when we call into this function?

Reply:
Yes we do but its called from updateTaskContainerResources() where the data interface is passed without casting to already exiting functions for lcow and wcow to update resources. So doing it in this function instead.